frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/index.tsx

```diff
@@ -207,79 +50,155 @@ const DatasetRow: FC<Props> = ({
 
-  useEffect(() => {
-    if (isLoading)
-      updateUploadProgress(
-        datasetStatus.upload_progress,
-        lastUploadProgress,
-        setLastUploadProgress,
-        invalidateCollectionQuery
-      );
+  const isNamePopulated = Boolean(dataset.name);
 
-    if (hasFailed.isFailed || nameIsPopulated) {
-      queryCache.cancelQueries([USE_DATASET_STATUS, dataset.id]);
-    }
-  }, [
-    dataset.id,
-    datasetStatus.upload_progress,
-    hasFailed.isFailed,
+  const name = dataset.name || file?.name || dataset.id;
+
+  // (thuang): We need to poll the collection until the name is populated,
+  // which indicates other metadata are populated too
+  useCheckCollectionPopulated({
     invalidateCollectionQuery,
-    isLoading,
-    lastUploadProgress,
-    nameIsPopulated,
+    isNamePopulated,
+    upload_progress,
+  });
+
+  const hasFailed = checkIfFailed(datasetStatus);
+  const { isFailed, error } = hasFailed;
+
+  useCancelDatasetStatusQuery({
+    datasetId: dataset.id,
+    isFailed,
+    isNamePopulated,
     queryCache,
-  ]);
+  });
+
+  useUploadProgress({
+    newUploadProgress: upload_progress,
+    setUploadProgress,
+    uploadProgress,
+  });
+
+  const isLoading = checkIfLoading(datasetStatus);
+
+  const {
+    tissue,
+    assay,
+    disease,
+    organism,
+    cell_count,
+  } = aggregateDatasetsMetadata([dataset]);
 
   return (
     <StyledRow>
       <DetailsCell>
         <TitleContainer>
           <Checkbox onChange={() => checkHandler(dataset.id)} />
           <div>{name}</div>
-          {errorTooltip(hasFailed)}
+          <ErrorTooltip isFailed={isFailed} error={error} />
         </TitleContainer>
-        {isLoading && renderUploadStatus(datasetStatus)}
+        {isLoading && (
+          <UploadStatus
+            isValidating={
+              datasetStatus.validation_status === VALIDATION_STATUS.VALIDATING
+            }
+            progress={datasetStatus.upload_progress}
+          />
+        )}
       </DetailsCell>
-      {conditionalPopover(tissue, isLoading, hasFailed.isFailed)}
-      {conditionalPopover(assay, isLoading, hasFailed.isFailed)}
-      {conditionalPopover(disease, isLoading, hasFailed.isFailed)}
-      {conditionalPopover(organism, isLoading, hasFailed.isFailed)}
-      {cellCount(cell_count, hasFailed.isFailed, isLoading)}
+      <Popover values={tissue} isLoading={isLoading} isFailed={isFailed} />
+      <Popover values={assay} isLoading={isLoading} isFailed={isFailed} />
+      <Popover values={disease} isLoading={isLoading} isFailed={isFailed} />
+      <Popover values={organism} isLoading={isLoading} isFailed={isFailed} />
+      <CellCount cellCount={cell_count} isLoading={isLoading} />
       <RightAlignedDetailsCell>
-        {nameIsPopulated && (
+        {isNamePopulated && (
           <Button intent={Intent.PRIMARY} outlined text="Explore" />
         )}
       </RightAlignedDetailsCell>
     </StyledRow>
   );
 };
 
+function useCheckCollectionPopulated({
```

Comment:
Is this supposed to be checkCollectionPopulated or checkDatasetPopulated?

Why are we invalidating the collection query on an interval?

Reply:
Oh I see, this is a workaround not having a signifier that we're done with the upload process and that the metadata is populated.

Once the backend is implemented, the useEffect will be called by an update to the datasetStatus and the interval won't be needed. Correct?

Reply:
I think BE is not planning to implement the complete status we were hoping for, and want us to just rely on checking for `isNamePopulated`!

Reply:
Wouldn't `cxg_conversion_status` be the substitute?

Reply:
I asked BE this question in [Slack](https://chanzuckerbergteam.slack.com/archives/C01CPJURT0S/p1609791352221300), and Marcus suggested `title` or I think he meant `name`, since it's a mandatory field:

https://chanzuckerbergteam.slack.com/archives/C01CPJURT0S/p1609795978226100?thread_ts=1609791352.221300&cid=C01CPJURT0S

So we can continue using `isNamePopulated` for checking if the dataset's metadata have been populated

For whether to show `Explore` button or not, that will be based on `cxg_conversion_status`!